server/events/approve_policies_command_runner.go

```diff
@@ -12,40 +12,58 @@ func NewApprovePoliciesCommandRunner(
 	commitStatusUpdater CommitStatusUpdater,
 	prjCommandBuilder ProjectApprovePoliciesCommandBuilder,
 	prjCommandRunner ProjectApprovePoliciesCommandRunner,
-	pullUpdater *PullUpdater,
+	commitOutputUpdater CommitOutputUpdater,
 	dbUpdater *DBUpdater,
 ) *ApprovePoliciesCommandRunner {
 	return &ApprovePoliciesCommandRunner{
 		commitStatusUpdater: commitStatusUpdater,
 		prjCmdBuilder:       prjCommandBuilder,
 		prjCmdRunner:        prjCommandRunner,
-		pullUpdater:         pullUpdater,
+		commitOutputUpdater: commitOutputUpdater,
 		dbUpdater:           dbUpdater,
 	}
 }
 
 type ApprovePoliciesCommandRunner struct {
 	commitStatusUpdater CommitStatusUpdater
-	pullUpdater         *PullUpdater
+	commitOutputUpdater CommitOutputUpdater
 	dbUpdater           *DBUpdater
 	prjCmdBuilder       ProjectApprovePoliciesCommandBuilder
 	prjCmdRunner        ProjectApprovePoliciesCommandRunner
 }
 
-func (a *ApprovePoliciesCommandRunner) Run(ctx *command.Context, cmd *command.Comment) {
+func (p *ApprovePoliciesCommandRunner) updateCombinedCommitStatusAndLogError(ctx *command.Context, status models.CommitStatus, cmdName fmt.Stringer) {
 	baseRepo := ctx.Pull.BaseRepo
 	pull := ctx.Pull
+	statusID, err := p.commitStatusUpdater.UpdateCombined(context.TODO(), baseRepo, pull, status, cmdName, ctx.StatusID)
+	if err != nil {
+		ctx.Log.Warnf("unable to update commit status: %s", err)
+	}
 
-	if err := a.commitStatusUpdater.UpdateCombined(context.TODO(), baseRepo, pull, models.PendingCommitStatus, command.PolicyCheck); err != nil {
+	// Assign the status ID to the commandContext for updates later in the workflow
+	ctx.StatusID = statusID
+}
+
+func (p *ApprovePoliciesCommandRunner) updateCombinedCountCommitStatusAndLogError(ctx *command.Context, status models.CommitStatus, cmdName fmt.Stringer, numSuccess int, numTotal int) {
+	baseRepo := ctx.Pull.BaseRepo
+	pull := ctx.Pull
+	statusID, err := p.commitStatusUpdater.UpdateCombinedCount(context.TODO(), baseRepo, pull, status, cmdName, numSuccess, numTotal, ctx.StatusID)
+	if err != nil {
 		ctx.Log.Warnf("unable to update commit status: %s", err)
 	}
 
+	// Assign the status ID to the commandContext for updates later in the workflow
+	ctx.StatusID = statusID
```

Comment:
same point about mutating objects, can't you just keep the status id outside the command context here if it's only used in the same file?